bugsnag-plugin-android-exitinfo/src/main/java/com/bugsnag/android/BugsnagExitInfoPlugin.kt

```diff
@@ -42,6 +45,26 @@ class BugsnagExitInfoPlugin @JvmOverloads constructor(
             ),
             exitInfoPluginStore
         )
+        InternalHooks.setEventStoreEmptyCallback(client) {
+            val eventSynthesizer = EventSynthesizer(
+                TraceEventEnhancer(
+                    client.logger,
+                    client.immutableConfig.projectPackages
+                ),
+                exitInfoPluginStore
+            )
+            val context = client.appContext
+            val am: ActivityManager =
+                context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+            val allExitInfo: List<ApplicationExitInfo> =
+                am.getHistoricalProcessExitReasons(context.packageName, 0, 100)
+            allExitInfo.forEach {
+                val newEvent = eventSynthesizer.createEventWithAnrExitInfo(it)
+                if (newEvent != null) {
+                    InternalHooks.delivery(client, newEvent)
+                }
+            }
```

Comment:
This block should possibly be moved to a utility function, since it's not really part of the `load` and adds noise